pkg/minishift/openshift/service.go

```diff
@@ -46,71 +51,79 @@ type Service struct {
 	} `json:"items"`
 }
 
-type Route struct {
+type RouteSpec struct {
 	Items []struct {
 		Spec struct {
 			AlternateBackends []struct {
 				Name   string `json:"name"`
 				Weight int    `json:"weight"`
 			} `json:"alternateBackends"`
 			Host string `json:"host"`
-			To   struct {
+			Tls  struct {
+				Termination string `json:"termination"`
+			} `json:"tls"`
+			To struct {
 				Name   string `json:"name"`
 				Weight int    `json:"weight"`
 			} `json:"to"`
 		} `json:"spec"`
 	} `json:"items"`
 }
 
-type ServiceWeight struct {
+type Route struct {
 	Name   string
 	Weight string
+	Tls    string
 }
 
-type ServiceSpec struct {
+type Service struct {
 	Namespace string
 	Name      string
 	URL       []string
 	NodePort  string
 	Weight    []string
 }
 
+type Services struct {
+	Service map[string]*Service
+}
+
 const (
 	ProjectsCustomCol = "-o=custom-columns=NAME:.metadata.name"
 )
 
-// GetServiceSpecs takes Namespace string and return route/nodeport/name/weight in a ServiceSpec structure
-func GetServiceSpecs(serviceNamespace string) ([]ServiceSpec, error) {
-	var serviceSpecs []ServiceSpec
+// GetServices takes Namespace string and return route/nodeport/name/weight in a Service structure
+func GetServices(serviceNamespace string) ([]Service, error) {
+	var services []Service
 
 	if serviceNamespace != "" && !isProjectExists(serviceNamespace) {
-		return serviceSpecs, errors.New(fmt.Sprintf("Namespace '%s' doesn't exist", serviceNamespace))
+		return services, errors.New(fmt.Sprintf("Namespace '%s' doesn't exist", serviceNamespace))
 	}
 
 	namespaces, err := getValidNamespaces(serviceNamespace)
 	if err != nil {
-		return serviceSpecs, err
+		return services, err
 	}
 
 	// iterate over namespaces, get command output, format route and nodePort
 	for _, namespace := range namespaces {
-		serviceNodePort, err := getService(namespace)
+		serviceNodePort, err := getNodePorts(namespace)
 		if err != nil {
-			return serviceSpecs, err
+			return services, err
 		}
-		routeSpec, err := getRouteSpec(namespace)
+		route, err := getRoutes(namespace)
 		if err != nil {
-			return serviceSpecs, err
+			return services, err
```

Comment:
Till this point services is empty, right? cc @anjannath

Reply:
@LalatenduMohanty Yes..